plugin/inc/installsteps/handleprofilerightsinstallstep.class.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+namespace GlpiPlugin\Iservice\InstallSteps;
+
+class HandleProfileRightsInstallStep
+{
+    const CLEANUP_ON_UNINSTALL = true;
+
+    public static function do(): bool
+    {
+        global $DB;
+
+        $table = 'glpi_profilerights';
+        $superAdminProfileId = \Profile::getSuperAdminProfilesId();
+        $rightsValues = include __DIR__ . DIRECTORY_SEPARATOR . '..' . DIRECTORY_SEPARATOR . 'rights.config.php';
+        $result = false;
+
+        if (!self::backupTable($table)) {
+            return false;
+        }
+
+        foreach ($rightsValues as $rightName => $rightValue) {
+            $result = $DB->update(
+                $table,
+                [
+                    'rights' => $rightValue,
+                ],
+                [
+                    'WHERE'  => [
+                        'name' => $rightName,
+                        'profiles_id' => ['!=', $superAdminProfileId],
+                    ],
+                ]
+            );
+
+            if ($result === false) {
+                return false;
+            }
+        }
+
+        return $result;
+    }
+
+    public static function undo(): void
+    {
+        if (!self::CLEANUP_ON_UNINSTALL) {
+            return;
+        }
+
+        global $DB;
+
+        $DB->runFile(PLUGIN_ISERVICE_DIR . '/install/sql/glpi_profilerights_original.sql');
+    }
+
+    public static function backupTable($tableName): bool
```

Comment:
This function does not back up a table, but creates a restore script for a table
```suggestion
    public static function createTableRestoreScript($tableName): bool
```